subprojects/execution/src/main/java/org/gradle/internal/execution/ExecutionResult.java

```diff
@@ -16,8 +16,11 @@
 
 package org.gradle.internal.execution;
 
+import javax.annotation.Nullable;
+
 public interface ExecutionResult {
```

Comment:
Can you add some Javadoc to this interface? Especially what it is about and when `getOutput()` is null?

Reply:
I moved this into `ExecutionEngine` and renamed it to `Execution`. I haven't added more Javadoc. Instead I started a new PR to generify `UnitOfWork` with the output type: https://github.com/gradle/gradle/pull/22066 I'm not yet sure if that approach is worth it: it adds a ton of noise in a number of places unfortunately.

I think it's okay to improve some things and not improve this type further in this PR. Let's come back to this later.

Reply:
Alright.